spring-rabbit/src/main/java/org/springframework/amqp/rabbit/config/RabbitNamespaceUtils.java

```diff
@@ -192,7 +192,7 @@ public static BeanDefinition parseContainer(Element containerEle, ParserContext
 		String channelTransacted = containerEle.getAttribute(CHANNEL_TRANSACTED_ATTRIBUTE);
 		if (StringUtils.hasText(channelTransacted)) {
 			// Note: a placeholder will pass this test, but if it resolves to true, it will be caught during container initialization
-			if (acknowledgeMode.isAutoAck() && channelTransacted.equalsIgnoreCase("true")) {
+			if (acknowledgeMode == AcknowledgeMode.NONE && channelTransacted.equalsIgnoreCase("true")) {
```

Comment:
We can flip condition: `AcknowledgeMode.NONE == acknowledgeMode`. 
This way we will protect ourselves from `NPE` as well.

Well, I'd say that we accept the PR, but I don't like that test-case. Especially the name mentioning `AMQP-274`.

@garyrussell , your opinion?